relm-derive/src/gen/parser.rs

```diff
@@ -227,39 +214,36 @@ impl RelmWidget {
     }
 }
 
-struct WidgetList {
-    widgets: Vec<Widget>,
+pub struct WidgetList {
+    pub widgets: Vec<Widget>,
 }
 
 impl Parse for WidgetList {
     fn parse(input: ParseStream) -> Result<Self> {
-        let widget: Widget = input.parse()?;
-        let mut widgets = vec![widget];
-        while !input.is_empty() {
-            let widget: Widget = input.parse()?;
-            widgets.push(widget);
-        }
-        Ok(WidgetList {
-            widgets,
-        })
-    }
-}
-
-pub fn parse_widgets(tokens: TokenStream) -> Result<Vec<Widget>> {
-    if let Ok(literal) = parse2::<LitStr>(tokens.clone()) {
-        // TODO: also support glade file.
-        let mut file = File::open(literal.value()).expect("File::open() in parse()");
-        let mut file_content = String::new();
-        file.read_to_string(&mut file_content).expect("read_to_string() in parse()");
-        let tokens = proc_macro::TokenStream::from_str(&file_content).expect("convert string to TokenStream");
-        let tokens = respan_with(tokens, literal.span().unwrap());
+        let lookahead = input.lookahead1();
+        if lookahead.peek(LitStr) {
+            let literal: LitStr = input.parse()?;
+
+            // TODO: also support glade file.
+            let mut file = File::open(literal.value()).expect("File::open() in parse()");
+            let mut file_content = String::new();
+            file.read_to_string(&mut file_content)
+                .expect("read_to_string() in parse()");
+            let tokens = proc_macro::TokenStream::from_str(&file_content)
+                .expect("convert string to TokenStream");
+            let tokens = respan_with(tokens, literal.span().unwrap());
+            return syn::parse(tokens);
```

Comment:
Since the `return` is last, please remove the `return` and the `;` at the end. Same for the other two that follows.